dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaConnectorConstants.java

```diff
@@ -26,6 +26,10 @@
     //The producer or consumer topic
     public static final String TOPIC = "topic";
 
+    //If the topic is set to this value then when the message payload is a ProducerRecord,
+    //use the topic set in the ProducerRecord rather than the one set in MP Config
+    public static final String USE_PRODUCER_RECORD_TOPIC = "USE_PRODUCER_RECORD_TOPIC";
```

Comment:
If we go ahead with something using this approach, we should make sure the special value makes sense to the user configuring the application.

I think something like this would make more sense:
```properties
mp.reactive.messaging.outgoing.foobar.topic=APPLICATION_PROVIDED
```

Reply:
Decided that we don't need this special config property value